render/canvas/src/lib.rs

```diff
@@ -545,17 +626,54 @@ impl RenderBackend for WebCanvasRenderBackend {
         // Noop
     }
 
-    fn render_bitmap(&mut self, bitmap: BitmapHandle, transform: &Transform, _smoothing: bool) {
+    fn render_bitmap(&mut self, bitmap: BitmapHandle, transform: &Transform, smoothing: bool) {
         if self.deactivating_mask {
             return;
         }
 
+        self.context.set_image_smoothing_enabled(smoothing);
+
         self.set_transform(&transform.matrix);
         self.set_color_filter(transform);
         if let Some(bitmap) = self.bitmaps.get(bitmap.0) {
-            let _ = self
-                .context
-                .draw_image_with_html_image_element(&bitmap.image, 0.0, 0.0);
+            match &bitmap.image {
+                BitmapDataStorage::HtmlElement(elem) => {
+                    let _ = self
+                        .context
+                        .draw_image_with_html_image_element(elem, 0.0, 0.0);
+                }
+                BitmapDataStorage::ImageData(data) => {
+                    let window = web_sys::window().unwrap();
+                    let document = window.document().unwrap();
+
+                    let canvas: HtmlCanvasElement = document
+                        .create_element("canvas")
+                        .unwrap()
+                        .dyn_into()
+                        .unwrap();
+
+                    canvas
+                        .style()
+                        .set_property("display", "none")
+                        .warn_on_error();
+
+                    canvas.set_width(bitmap.width);
+                    canvas.set_height(bitmap.height);
+
+                    let context: CanvasRenderingContext2d = canvas
+                        .get_context("2d")
+                        .unwrap()
+                        .unwrap()
+                        .dyn_into()
+                        .unwrap();
+
+                    let _ = context.put_image_data(data, 0.0, 0.0).unwrap();
```

Comment:
Could we perhaps cache this canvas with the image drawn as well, instead of re-creating on each render?